### PR TITLE
JAMES-2453 Remove JUNIT 4 usages in server/protocols

### PR DESCRIPTION
**Pull request by chibenwa**
(Except soon-to-be-deprecated JMAP draft)

Now we remain:

```
$ grep 'import org.junit.Test;' * -R | wc -l
357
```

119 being caused by JMAP draft

### CODE REVIEW

File: third-party/spamassassin/src/test/java/org/apache/james/spamassassin/mock/MockSpamdExtension.java [hunk 1]
```diff
@@ -23,25 +23,29 @@
...
 
 import org.apache.james.util.concurrent.NamedThreadFactory;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
 import org.junit.rules.ExternalResource;
 
-public class MockSpamdTestRule extends ExternalResource {
+public class MockSpamdExtension implements AfterEachCallback, BeforeEachCallback {
```
Comment (jeantil) [inline third-party/spamassassin/src/test/java/org/apache/james/spamassassin/mock/MockSpamdExtension.java:32]: the last of the public modifiers

I know how painful it is to do this, I won't blame you if you don't want to do it :)

**Review by jeantil — APPROVED** [review]
Another step in the right direction thanks !

As said in my last comment, I flagged the places where the public modifiers were not removed but I know how painful it is to go all over again to remove them. Don't feel compelled to do it.

File: third-party/spamassassin/src/test/java/org/apache/james/spamassassin/mock/MockSpamdExtension.java (hunk 1 above)
Comment (Arsnael) [reply third-party/spamassassin/src/test/java/org/apache/james/spamassassin/mock/MockSpamdExtension.java:32]: Not this one, it's an extension declaration that will be used in oither Test classes `SpamAssassinTest` and `SpamAssassinHandlerTest` :)

**Comment by Arsnael** [comment]
Let me know if your concerns got answered @jeantil :)

**Review by jeantil — APPROVED** [review]
looks good to me !

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**